build.py

```diff
@@ -1716,30 +1726,43 @@ def __check_tools(self, opts):
             error_exit("%s not found. Please check that you installed wget in msys2 using ``pacman -S wget``" % (self.wget,))
         print_debug("wget: %s" % (self.wget,))
 
+    def add_env(self, key, value, prepend=True):
+        env = os.environ
+        te = env.get(key, None)
+        if te:
+            if prepend:
+                env[key] = value + ';' + te
+            else:
+                env[key] = te + ';' + value
+        else:
+            # not set
+            env[key] = value
+
     def __check_vs(self, opts):
         # Verify VS exists at the indicated location, and that it supports the required target
-        if opts.platform in ('Win32', 'win32', 'x86'):
-            opts.platform = 'Win32'
-            self.filename_arch = 'x86'
+        if opts.platform == 'Win32':
             vcvars_bat = os.path.join(opts.vs_install_path, 'VC', 'bin', 'vcvars32.bat')
-        elif opts.platform in ('x64', 'amd64', 'Amd64'):
-            opts.platform = 'x64'
-            self.filename_arch = 'x64'
+        else:
             vcvars_bat = os.path.join(opts.vs_install_path, 'VC', 'bin', 'amd64', 'vcvars64.bat')
             # make sure it works even with VS Express
             if not os.path.exists(vcvars_bat):
                 vcvars_bat = os.path.join(opts.vs_install_path, 'VC', 'bin', 'x86_amd64', 'vcvarsx86_amd64.bat')
-        else:
-            raise Exception("Invalid target platform '%s'" % (opts.platform,))
 
         if not os.path.exists(vcvars_bat):
             raise Exception("'%s' could not be found. Please check you have Visual Studio installed at '%s' and that it supports the target platform '%s'." % (vcvars_bat, opts.vs_install_path, opts.platform))
 
+        # Add to the environment the gtk paths so meson can find everything
+        self.add_env('INCLUDE', self.gtk_dir + r'\INCLUDE')

```

Comment:
It's too late, I need to rest. The path (r'\INCLUDE') should be lower case, not the set variable, right ?